pkg/operator/configobserver/oauth/observe_tokenconfig.go

```diff
@@ -0,0 +1,191 @@
+package oauth
+
+import (
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+
+	"github.com/imdario/mergo"
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+// OAuthLister lists OAuth information and allows resources to be synced
+type OAuthLister interface {
+	OAuthLister() configlistersv1.OAuthLister
+	PreRunHasSynced() []cache.InformerSynced
+}
+
+var (
+	tokenConfigPath = []string{"oauthConfig", "tokenConfig"}
+)
+
+const (
+	defaultAccessTokenMaxAgeSeconds          = float64(86400) // a day
+	fieldAccessTokenMaxAgeSeconds            = "accessTokenMaxAgeSeconds"
+	fieldAccessTokenInactivityTimeoutSeconds = "accessTokenInactivityTimeoutSeconds"
+)
+
+// ObserveTokenConfig returns the merged config for access token max age and access token inactivity timeout.
+func ObserveTokenConfig(genericlisters configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	errs = []error{}
+	mergedConfig := map[string]interface{}{}
+
+	for _, observeConfig := range []configobserver.ObserveConfigFunc{ObserveAccessTokenMaxAgeSeconds, ObserveAccessTokenInactivityTimeoutSeconds} {
+		observedConfig, err := observeConfig(genericlisters, recorder, existingConfig)
+		if err != nil {
+			errs = append(errs, err...)
+		}
+
+		if err := mergo.Merge(&mergedConfig, observedConfig); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	return mergedConfig, errs
+}
+
+// ObserveAccessTokenMaxAgeSeconds returns the observed config for access token max age.
+// If the function fails to read both AccessTokenMaxAgeSeconds from existing config and OAuth cluster config,
+// it uses the default value for token max age.
+// If the function only fails to read OAuth cluster config, it returns existing config.
+// If token max age value is zero in OAuth cluster config, it uses default value.
+// In any other case, it uses the value present in OAuth cluster config.
+func ObserveAccessTokenMaxAgeSeconds(genericlisters configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
```

Comment:
(No action required) I think it might be simpler to separate mixing value retrieval and return value construction. Maybe separate out retrieval of the value from oauth config into a separate function?

Reply:
I will look into this.